cloud-gateway-service/src/main/java/org/zowe/apiml/cloudgatewayservice/service/WebClientHelper.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+
+package org.zowe.apiml.cloudgatewayservice.service;
+
+import io.netty.handler.ssl.SslContext;
+import io.netty.handler.ssl.SslContextBuilder;
+import io.netty.handler.ssl.util.InsecureTrustManagerFactory;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.zowe.apiml.message.log.ApimlLogger;
+import org.zowe.apiml.message.yaml.YamlMessageServiceInstance;
+import org.zowe.apiml.security.HttpsConfigError;
+
+import javax.net.ssl.KeyManagerFactory;
+import javax.net.ssl.SSLException;
+import java.io.File;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.UnrecoverableKeyException;
+
+/**
+ * Utility class for the custom Netty {@link SslContext} creation.
+ * Does not support keyring because client keystore override mainly used in development mode and not supposed be run on the Mainframe.
+ */
+@Slf4j
+@NoArgsConstructor(access = AccessLevel.PROTECTED)
+public class WebClientHelper {
+    private static final ApimlLogger apimlLog = ApimlLogger.of(WebClientHelper.class, YamlMessageServiceInstance.getInstance());
+
+    /**
+     * Load {@link SslContext} from the specified keystore
+     *
+     * @param keystorePath path to the keystore file
+     * @param password     keystore password
+     * @throws IllegalArgumentException if keystore file does not exist.
+     * @throws HttpsConfigError         if any error occur during the context creation.
+     */
+    public static SslContext load(String keystorePath, char[] password) {
+        File keyStoreFile = new File(keystorePath);
+        if (keyStoreFile.exists()) {
+            try (InputStream is = Files.newInputStream(Paths.get(keystorePath))) {
+                KeyStore keyStore = KeyStore.getInstance("PKCS12");
```

Comment:
what is the reason of not taking this value from configuration?

Reply:
Here I created empty keystore instance in memory which will be loaded from another keystore in the file. As `PKCS12` standard for every java and x509 standard - it is safe to use it at the same type if is standard, default and does not require any code. Why keystores types should much each other?
[Docs](https://docs.oracle.com/javase/8/docs/api/java/security/KeyStore.html)

Reply:
JCERACFKS, JCECCARACFKS , JCEKS, I am not sure how it works internally within SPI but in other parts of the code, it is being specified as a type when calling this method.

Reply:
Ok, I've added one more custom property:`apiml.cloudGateway.clientKeystoreType`